pkg/reconciler/coordination/deployment/deployment_controller_test.go

```diff
@@ -0,0 +1,200 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kcp-dev/kcp/pkg/reconciler/committer"
+	"github.com/kcp-dev/kcp/tmc/pkg/coordination"
+)
+
+func intPtr(i int32) *int32 {
+	res := i
+	return &res
+}
+
+type mockedPatcher struct {
+	clusterName  logicalcluster.Name
+	namespace    string
+	appliedPatch string
+}
+
+func (p *mockedPatcher) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (*Deployment, error) {
+	p.appliedPatch = string(data)
+	return nil, nil
+}
+
+func TestUpstreamViewReconciler(t *testing.T) {
+	tests := map[string]struct {
+		input        *appsv1.Deployment
+		appliedPatch string
+		wantError    bool
+	}{
+		"spread requested replicas": {
+			input: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "test",
+					UID:             types.UID("uid"),
+					ResourceVersion: "resourceVersion",
+					Labels: map[string]string{
+						"state.workload.kcp.dev/syncTarget1": "Sync",
+						"state.workload.kcp.dev/syncTarget2": "Sync",
+					},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: intPtr(7),
+				},
+			},
+			appliedPatch: `{"metadata":{"annotations":{"experimental.spec-diff.workload.kcp.dev/syncTarget1":"[{ \"op\": \"replace\", \"path\": \"/replicas\", \"value\": 4 }]","experimental.spec-diff.workload.kcp.dev/syncTarget2":"[{ \"op\": \"replace\", \"path\": \"/replicas\", \"value\": 3 }]"},"resourceVersion":"resourceVersion","uid":"uid"}}`,
+		},
+		"don't take empty labels into account": {
+			input: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "test",
+					UID:             types.UID("uid"),
+					ResourceVersion: "resourceVersion",
+					Labels: map[string]string{
+						"state.workload.kcp.dev/syncTarget1": "Sync",
+						"state.workload.kcp.dev/syncTarget2": "",
+					},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: intPtr(7),
+				},
+			},
+			appliedPatch: `{"metadata":{"annotations":{"experimental.spec-diff.workload.kcp.dev/syncTarget1":"[{ \"op\": \"replace\", \"path\": \"/replicas\", \"value\": 7 }]"},"resourceVersion":"resourceVersion","uid":"uid"}}`,
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			var patcher *mockedPatcher
+			controller := controller{
+				getDeployment: func(lclusterName logicalcluster.Name, namespace, name string) (*appsv1.Deployment, error) {
+					return tc.input, nil
+				},
+				patcher: func(clusterName logicalcluster.Name, namespace string) committer.Patcher[*appsv1.Deployment] {
+					patcher = &mockedPatcher{
+						clusterName: clusterName,
+						namespace:   namespace,
+					}
+					return patcher
+				},
+				syncerViewRetriever: coordination.NewDefaultSyncerViewManager[*appsv1.Deployment](),
+			}
+
+			err := controller.processUpstreamView(context.Background(), "")
+			if tc.wantError {
+				require.Error(t, err)
```

Comment:
You'd usually want to `return` after this, to avoid asserting the "happy path" results. But it doesn't look like you ever set `wantError` to true? Maybe remove it?

Reply:
You're right that I never set it. Let me add some tests that expect it.
Abot the first half of your comment, I'm not sure see what you mean in the specific case. I don't see the difference with https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/apis/apibinding/apibinding_reconcile_test.go#L508 for example